frontend/services/api.ts

```diff
@@ -21,12 +21,16 @@ const API = axios.create({
 
 const onResponseSuccess = (response) => {
   try {
-    const parsedData = JSON.parse(response);
     return {
-      data: dataFormatter.deserialize(parsedData),
-      meta: parsedData.meta,
+      data: {
+        // Deserialized data
+        data: !!response.data && dataFormatter.deserialize(response.data),
+        // Metadata (pagination) if any
+        ...(!!response.data.meta && { meta: response.data.meta }),
```

Comment:
I think this may as well crash if trying to destructure `false`:
```suggestion
        ...(response.data.meta ? { meta: response.data.meta } : {}),
```